app/code/core/Mage/Core/Block/Messages.php

```diff
@@ -247,6 +247,7 @@ public function getGroupedHtml()
         if ($html) {
             $html .= '</' . $this->_messagesFirstLevelTagName . '>';
         }
+        $this->_messages = null;
```

Comment:
PHPStan is not happy with this statement.

Reply:
In my case I push the wrong button and I approved this PR. It needs more 2 green approves.

The reported issue I encountered a few times over the years.